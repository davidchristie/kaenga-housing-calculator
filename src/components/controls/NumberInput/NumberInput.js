import PropTypes from 'prop-types'
import React, { Component } from 'react'
import {
  ControlLabel,
  FormControl,
  FormGroup,
  HelpBlock
} from 'react-bootstrap'

class NumberInput extends Component {
  render () {
    const {
      input: { onChange, value },
      label,
      max,
      meta: { error },
      min
    } = this.props
    return (
      <FormGroup>
        <ControlLabel>
          {label}
        </ControlLabel>
        <FormControl
          componentClass='input'
          max={max}
          min={min}
          onChange={onChange}
          type='number'
          value={value}
        />
        {error && <HelpBlock>{error}</HelpBlock>}
      </FormGroup>
    )
  }
}

NumberInput.defaultProps = {
  input: {},
  label: '',
  meta: {},
  min: 0
}

NumberInput.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.any
  }).isRequired,
  label: PropTypes.string,
  max: PropTypes.number,
  meta: PropTypes.shape({
    error: PropTypes.string
  }).isRequired,
  min: PropTypes.number
}

export default NumberInput
