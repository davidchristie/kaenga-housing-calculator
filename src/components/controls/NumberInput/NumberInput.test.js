import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'

import NumberInput from './NumberInput'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(
    <NumberInput />
  )).toMatchSnapshot()
})
