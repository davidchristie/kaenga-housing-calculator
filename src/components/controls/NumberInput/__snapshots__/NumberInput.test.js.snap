exports[`test matches snapshot 1`] = `
<FormGroup
  bsClass="form-group">
  <ControlLabel
    bsClass="control-label"
    srOnly={false} />
  <FormControl
    bsClass="form-control"
    componentClass="input"
    max={undefined}
    min={0}
    onChange={undefined}
    type="number"
    value={undefined} />
</FormGroup>
`;
